lib/runtime-core/src/typed_func.rs

```diff
@@ -438,33 +472,91 @@ macro_rules! impl_traits {
             }
         }
 
-        impl< $( $x, )* Rets, Trap, FN > ExternalFunction<( $( $x ),* ), Rets> for FN
+        impl< $( $x, )* Rets, Trap, FN > ExternalFunction<ExplicitVmCtx, ( $( $x ),* ), Rets> for FN
+        where
+            $( $x: WasmExternType, )*
+            Rets: WasmTypeList,
+            Trap: TrapEarly<Rets>,
+            FN: Fn(&mut vm::Ctx $( , $x )*) -> Trap,
+        {
+            #[allow(non_snake_case)]
+            fn to_raw(&self) -> NonNull<vm::Func> {
+                if mem::size_of::<Self>() == 0 {
+                    /// This is required for the llvm backend to be able to unwind through this function.
+                    #[cfg_attr(nightly, unwind(allowed))]
+                    extern fn wrap<$( $x, )* Rets, Trap, FN>(
+                        vmctx: &mut vm::Ctx $( , $x: <$x as WasmExternType>::Native )*
+                    ) -> Rets::CStruct
+                    where
+                        $( $x: WasmExternType, )*
+                        Rets: WasmTypeList,
+                        Trap: TrapEarly<Rets>,
+                        FN: Fn(&mut vm::Ctx, $( $x, )*) -> Trap,
+                    {
+                        let f: FN = unsafe { mem::transmute_copy(&()) };
```

Comment:
This line is terrifying and deserves some comments.  Also is `()` ABI safe? 🤔

Reply:
This is the original code. I'm happy to discuss about this details but I prefer to address it in another PR. I'm more inclined to split the modifications of this part of the code in small PR. Thoughts?

Reply:
Additional note: The `wrap` function is almost a duplication: Only the `FN` definition is different, along with the way `f` is called (with or without `vmctx`).

Reply:
If the code already existed, we can leave it for now.  I have no idea how it works though!

Reply:
:+1: